src/map/status.cpp

```diff
@@ -5412,6 +5412,11 @@ void status_calc_state( struct block_list *bl, status_change *sc, std::bitset<SC
 				  || (sc->getSCE(SC_CRYSTALIZE) && bl->type != BL_MOB)
  				 )
 				 sc->cant.move += (start ? 1 : ((sc->cant.move) ? -1 : 0));
+#ifndef RENEWAL
+		// Remove movement restriction when Longing for Freedom becomes active with an Ensemble skill.
+		if (start && sc->getSCE(SC_DANCING) && sc->getSCE(SC_DANCING)->val4 && sc->getSCE(SC_LONGING))
+			sc->cant.move = 0;
```

Comment:
Shouldn't it be just substracting "1" in case you also can't move out of other reasons?

Reply:
If I was not mistaken this would also be fixed with #8834 btw.
There I fixed the counting.

Reply:
True, but it can take a long time until that can go live, no? Because of the memory access issue.
Wouldn't it be better to fix this one for now?
Though the build doesn't seem to be working anymore...

Reply:
Well, i comment just to give my experience last year when i tried this pull request on my own server.
If it's outdated with the new src of rAthena does not take into account my comment because I have made a huge pause on RO, As I said, when I tried this one, my character has able to move inside the song, the only thing you can't is to attacked in close combat, you can use others skill like Arrow vulcan but if you want to attack in close combat with normal attacks, you have to wait for the song ending.
There is a video on my old comment if you want what is going on.
Sorry for my bad english, I hope it will be understandable for you guys !